From 7227319b0abcb7c3a2cc2547677ec49f2c648c72 Mon Sep 17 00:00:00 2001
From: Arunachalam Ganapathy <arunachalam.ganapathy@arm.com>
Date: Fri, 31 May 2024 15:05:40 +0100
Subject: [PATCH] force MBEDTLS_CONFIG_FILE definition

Mbed TLS in RMM must be compiled with a custom config file.

Signed-off-by: Arunachalam Ganapathy <arunachalam.ganapathy@arm.com>
---
 include/mbedtls/build_info.h | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/include/mbedtls/build_info.h b/include/mbedtls/build_info.h
index eab167f38..ab5bcd3cb 100644
--- a/include/mbedtls/build_info.h
+++ b/include/mbedtls/build_info.h
@@ -103,7 +103,13 @@
 
 /* X.509, TLS and non-PSA crypto configuration */
 #if !defined(MBEDTLS_CONFIG_FILE)
-#include "mbedtls/mbedtls_config.h"
+/*
+ * Mbed TLS in RMM must be compiled with custom config file. This causes the
+ * build to fail if MBEDTLS_CONFIG_FILE is not defined. Any source in RMM or
+ * 3rd party projects that includes <mbedtls/build_info.h>, must define
+ * MBEDTLS_CONFIG_FILE.
+ */
+#error "MBEDTLS_CONFIG_FILE not defined"
 #else
 #include MBEDTLS_CONFIG_FILE
 #endif
-- 
2.39.2

